unix-domain-socket/src/test/java/akka/stream/alpakka/unixdomainsocket/javadsl/UnixDomainSocketTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.unixdomainsocket.javadsl;
+
+import akka.NotUsed;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.Framing;
+import akka.stream.javadsl.FramingTruncation;
+import akka.stream.javadsl.Source;
+import akka.util.ByteString;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import akka.actor.*;
+import akka.stream.*;
+import akka.testkit.*;
+import akka.japi.Pair;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.concurrent.CompletionStage;
+
+import akka.stream.alpakka.unixdomainsocket.javadsl.UnixDomainSocket.IncomingConnection;
+import akka.stream.alpakka.unixdomainsocket.javadsl.UnixDomainSocket.ServerBinding;
+
+public class UnixDomainSocketTest {
+
+    private static ActorSystem system;
+    private static Materializer materializer;
+
+    private static Pair<ActorSystem, Materializer> setupMaterializer() {
+        final ActorSystem system = ActorSystem.create();
+        final Materializer materializer = ActorMaterializer.create(system);
+        return Pair.create(system, materializer);
+    }
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        final Pair<ActorSystem, Materializer> sysmat = setupMaterializer();
+        system = sysmat.first();
+        materializer = sysmat.second();
+    }
+
+    @AfterClass
+    public static void teardown() {
+        JavaTestKit.shutdownActorSystem(system);
+    }
+
+    @Test
+    public void aUnixDomainSocketShouldReceiveWhatIsSent() throws Exception {
```

Comment:
This test is mainly provided to test compilation. I've not fleshed it out fully given its delegation to the Scala test. The Scala test has reasonable coverage.